experimental/design-notes.txt

```diff
@@ -0,0 +1,112 @@
+I started comparing OpenTelemetry metrics to other metric APIs and wanted to jot down observations and questions that came up along the way:
+
+Micrometer (MM) - https://micrometer.io/docs/concepts
+
+1. Do the names that OpenTelemetry chose match industry conventions? 
+For the MM library there are different types of Meters such as timers, counters, gauges, etc. Depending on their type, meters implement operations such
+as increment() and set(). Meters are associated with a single set of tags.
+For the OT spec Meters are factories that create instruments and it is the instruments that have the data collecting APIs such as Add() or Record(). Instruments
+aren't associated with a specific set of tags (labels) unless they are first bound.
+As best I can tell the best conceptual mapping is that MM's Meter == OT's bound instrument. It doesn't worry me too much that OT would use a different
+name for the same concept, what worries me more is if OT is reusing the same names but altering the concept they apply to.
+
+2. MM has no equivalent abstraction for OT's Meter or Instrument. It goes straight from MeterProvider (MeterRegistry in MM-terminology) to a bound-instrument 
+(Meter in MM-terminology). There is an even simpler case for people using a default global registry where they can use static functions defined in the Metrics class.
+If MM has been succesful with these simpler abstractions it raises the question what value is OT getting from its more complex 4 layers of
+abstraction (MeterProvider->Meter->Instrument->BoundInstrument)?
+
+3. MM landed on a different set of instrument types than OT did. It would be good to understand why we believe OT's instrument set is the right set to pick?
```

Comment:
Yeah, I think the OT SIG also pointed this out.  I think Reiley will likely drive this to some better conclusion.